.github/workflows/main.yaml

```diff
@@ -65,83 +65,14 @@ jobs:
-            engine: nerdctl
-          - lima-template: template://centos-stream-9
-            engine: podman
-          - lima-template: template://fedora
-            engine: podman
-    env:
-      LIMA_TEMPLATE: "${{ matrix.lima-template }}"
-      CONTAINER_ENGINE: "${{ matrix.engine }}"
-    steps:
-      - uses: actions/checkout@v4
-      - name: "Install QEMU"
-        run: |
-          set -eux
-          sudo apt-get update
-          sudo apt-get install -y --no-install-recommends ovmf qemu-system-x86 qemu-utils
-          sudo modprobe kvm
-          # `sudo usermod -aG kvm $(whoami)` does not take an effect on GHA
-          sudo chown $(whoami) /dev/kvm
+    name: "Multi node with defaults"
+    uses: ./.github/workflows/reusable-multi-node.yaml
 
-      - name: "Install Lima"
-        env:
-          GITHUB_TOKEN: ${{ github.token }}  # required by `gh attestation verify`
-        run: |
-          set -eux
-          LIMA_VERSION=$(curl -fsSL https://api.github.com/repos/lima-vm/lima/releases/latest | jq -r .tag_name)
-          FILE="lima-${LIMA_VERSION:1}-Linux-x86_64.tar.gz"
-          curl -fOSL https://github.com/lima-vm/lima/releases/download/${LIMA_VERSION}/${FILE}
-          gh attestation verify --owner=lima-vm "${FILE}"
-          sudo tar Cxzvf /usr/local "${FILE}"
-          rm -f "${FILE}"
-
-      - name: "Cache ~/.cache/lima"
-        uses: actions/cache@v4
-        with:
-          path: ~/.cache/lima
-          key: lima-${{ env.LIMA_VERSION }}
-
-      - name: "Relax disk pressure limit"
-        run: |
-          set -x
-          sudo snap install yq
-          yq -i 'select(.kind=="KubeletConfiguration").evictionHard."imagefs.available"="3Gi"' kubeadm-config.yaml
-      - run: ./hack/create-cluster-lima.sh
-      - run: kubectl taint nodes --all node-role.kubernetes.io/control-plane- || true
-      - run: ./hack/test-smoke.sh
-      - if: failure()
-        name: "kubectl get nodes"
-        run: |
-          set -x
-          kubectl get nodes -o wide
-          kubectl get nodes -o yaml
-          limactl shell host0 df -h
-          limactl shell host1 df -h
-      - if: failure()
-        name: "kubectl get pods"
-        run: |
-          set -x
-          kubectl get pods -A -o yaml
-          limactl shell host0 journalctl --user --no-pager --since "10 min ago"
-      - name: "Test data persistency after restarting the node"
-        run: |
-          limactl stop host0
-          limactl stop host1
-          limactl start host0
-          limactl start host1
-          # The plain mode of Lima disables automatic port forwarding
-          ssh -q -f -N -L 6443:127.0.0.1:6443 -F ~/.lima/host0/ssh.config lima-host0
-          sleep 30
-          ./hack/test-smoke.sh
+  multi-node-custom-port:
+    name: "Multi node with custom kube-apiserver port"
+    uses: ./.github/workflows/reusable-multi-node.yaml
+    with:
+      # Defaults to 6443
+      kube_apiserver_port: "8080"
```

Comment:
This should change all the ports, not just apiserver

Reply:
@AkihiroSuda each of these needs to be done separately and carefully, and I think in different PRs. Based on the challenges I hit with just api server, it would increase the scope here enormously, which already is more than I intended. We are going to need all of these services customized in the next fiscal year so you can count on me to come back and do the changes. My preference is to do one at a time and have clearly scoped PRs. If you want me to just add to the template here I could, but it would be misleading because the current setup does not support that customization.

Reply:
This is now done, with the exception of flannel, which is based on vxlan / a Linux default. See my comment in https://github.com/rootless-containers/usernetes/issues/345#issuecomment-2525465642

Reply:
Flannel is added.